src/components/NewClusterModal.js

```diff
@@ -197,16 +199,23 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
       options: _.map(({ label, image }) => ({ label, value: image }), leoImages)
     })
 
+    const isCustomImageInvalid = customEnvImage && !imageValidationRegexp.test(customEnvImage)
+    const isSelectedImageInputted = selectedLeoImage === CUSTOM_MODE || selectedLeoImage === PROJECT_SPECIFIC_MODE
+
     const makeGroupedEnvSelect = id => h(GroupedSelect, {
       id,
+      maxHeight: '25rem',
       value: selectedLeoImage,
       onChange: ({ value }) => {
-        this.setState({ selectedLeoImage: value })
+        this.setState({ selectedLeoImage: value, customEnvImage: '' })
```

Comment:
This means that if you switch to a prebuilt image and then back to custom, it loses any url you entered. Is that what we want?